### ISSUE TITLE
Stable Diffusion WebUI hangs on update

### ISSUE THREAD

**Issue by evanerichards** [LykosAI/StabilityMatrix#335]
I'm trying to update from 1.6.0 to 1.7.0 but it just spins infinitely (I assume. I haven't waited an infinitely long time yet).

It seems to freeze on the first line: HEAD is now at cf2772fa Merge branch 'release_candidate'

I'm on windows 10, Stability Matrix version 2.7.2

I'm not sure what other relevant information to include

*edit* actually, when I launch Stable Diffusion WebUI and it starts it shows that I'm running 1.7.0 even though the packages page still says v1.6.0 and has the button to update.

**Comment by agarzon** [comment]
Having the same issue, is stuck there 

![image](https://github.com/LykosAI/StabilityMatrix/assets/339828/f926d841-d505-4408-86b0-badb4ad20b63)

**Comment by evanerichards** [comment]
> Having the same issue, is stuck there
> 
> ![image](https://private-user-images.githubusercontent.com/339828/291021653-f926d841-d505-4408-86b0-badb4ad20b63.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI3NzY5MjIsIm5iZiI6MTcwMjc3NjYyMiwicGF0aCI6Ii8zMzk4MjgvMjkxMDIxNjUzLWY5MjZkODQxLWQ1MDUtNDQwOC04NmIwLWJhZGI0YWQyMGI2My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxN1QwMTMwMjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kYzM1M2Y1NzBlY2M1MGM3MTdiZTZmM2Q0MzViYTcwOTc4NTY3Y2Y3NTQzZmEzMjY2Nzc2NmJkOWNiMmNmM2JmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.3JJeYmQeQBtz6iFfYLYNr97RwnNRGtBumiksqQaV-o4)

Yes!! Check to see if it actually installed but just isn't reporting the update in the packages page.

**Comment by ionite34** [comment]
What version of Stability Matrix and OS were you on when this happened?

**Comment by agarzon** [comment]
> What version of Stability Matrix and OS were you on when this happened?

The latest 2.7.2, and still unable to update A1111

**Comment by unknownentity123** [comment]
Same issue here this evening when attempting to update

**Comment by KINGLIFER** [comment]
And guys do not bother their discord. I found 3 issues and was gaslighted that I must be doing something wrong and I should move folders or in the wrong sub... like....
I get it... this is free but you cant just break things in the "stable update". Had to uninstall.
Messed up the ipadapters in comfyui AGAIN.... I mean... just ridiculous.

**Comment by drdancm** [comment]
Same problem, update was running for over 7 hours and still not finished. When I checked I was at the same place:

_HEAD is now at cf2772fa Merge branch 'release_candidate'_

**Comment by ZedOud** [comment]
I also had this issue. 
I actually waited to update to 1.7.0 until today assuming Stability Matrix might need to make an update to accommodate, as most minor version changes are due to accommodations for other package updates.
Nope, nothing, so I assumed it was safe.
Obviously something is wrong with Stability Matrix.

**Comment by mohnjiles** [comment]
Sorry folks, still trying to get to the root of the issue, since we are unable to reproduce it ourselves. When it's fixed we'll close this issue & put a patch note in the changelog for it. 

One question, is anyone having this issue also connected to LDAP / logged into a domain / using their work computer? We suspect it may be related to a bug in the PortableGit we use.

In the meantime, if anyone would like to try [downloading the latest PortableGit](https://github.com/git-for-windows/git/releases/download/v2.43.0.windows.1/PortableGit-2.43.0-64-bit.7z.exe), run the download and tell it to extract to `<your_sm_install_dir>\PortableGit`, and see if that fixes the issue. 

Otherwise, like @ZedOud mentioned, it's just a display error and the package should have updated. If the console output shows `Version: v1.7.0`, you can edit your `settings.json` file and change any `v1.6.0` to `v1.7.0` to get rid of the update button.

**Comment by ZedOud** [comment]
I'll add that actually, Stable Diffusion doesn't actually finish launching for me, I can't access the local WebUI. But otherwise the launch look like it completed correctly and loaded the last model.

**Comment by ZedOud** [comment]
> In the meantime, if anyone would like to try [downloading the latest PortableGit](https://github.com/git-for-windows/git/releases/download/v2.43.0.windows.1/PortableGit-2.43.0-64-bit.7z.exe), run the download and tell it to extract to `<your_sm_install_dir>\PortableGit`, and see if that fixes the issue.

I did this. And hit "update". The dialogue simply says:
`HEAD is now at cf2772fa Merge branch 'release_candidate'`
and hangs.

**Comment by mohnjiles** [comment]
> I did this. And hit "update". The dialogue simply says: `HEAD is now at cf2772fa Merge branch 'release_candidate'` and hangs.

Thanks for trying! Back to the drawing board then...

Any additional system info you (or others feel free too) care to share? Anything to help pin down the issue...

**Comment by hyppyhyppo** [comment]
I'm on version 2.7.4, installed as portable so I moved it to another folder and have the same issue (update hangs on) but when launching A1111 it says version 1.7.0 and it's working fine.
By trying to hit update again the dialogue said:

From https://github.com/AUTOMATIC1111/stable-diffusion-webui

   a8e41f58..945cb979  gradio4       -> origin/gradio4

 * [new branch]        minor-xyz-fix -> origin/minor-xyz-fix

 * [new branch]        reduce-unnecessary-ui-config-write -> origin/reduce-unnecessary-ui-config-write

 * [new branch]        reorder-post-processing-modules -> origin/reorder-post-processing-modules

HEAD is now at cf2772fa Merge branch 'release_candidate'

**Comment by ZedOud** [comment]
After restarting my computer, the WebUI does function. However, I'm not sure what may have caused it to now work because I only restarted my computer after:

1. doing the 1.7.0 update and 
2. downloading the PortableGit version and hitting "update" again (the result of which I reported above)
3. Stable Diffusion's webui did not open/work

I then restarted my computer at this point, as well as played around with the other packages before checking again to see if SD's WebUI would work.
SD's WebUI did work at this point and I then generated an image in SD's WebUI and closed it and attempted to run the update again, this time there was slightly more in the details (annoyingly, I could highlight this text, but not copy it, so I OCR'd it with my phone):

`From https://github.com/AUTOMATIC1111/stable-diffusion-webui
92b33344..945cb979 gradio4 -> origin/gradio4
HEAD is now at cf2772fa Merge branch 'release_candidate'`

This update attempt also hanged. 
On restarting Stability Matrix and starting Stable Diffusion it once again ran through the normal starting procedure but refused to open the webui. (I did try closing and opening Stable Diffusion a few times, but nothing changed.)
Once again, I restarted my computer, and upon starting Stability Matrix and Stable Diffusion its webui worked and I was able to generate an image.


So not completely resolved as the "update" button and stated update version is still in error, and hitting "update" breaks it somehow until you restart, but at least the Stable Diffusion installation functions.

**Comment by Kryptonic83** [comment]
Didn't get a response from discord, so posting here as well.
My a1111 isn't updating from 1.6 to 1.7 (not just visual bug, still shows 1.6 in console and webui), check for updates doesn't seem to do anything
[](https://i.imgur.com/HiGn38U.png)
[](https://i.imgur.com/tB2XMAS.png)
[](https://i.imgur.com/f5IPEzU.png)
I do have comfyui and here's what my controlnet folders look like https://i.imgur.com/jPamExy.png
[](https://i.imgur.com/VL0nRCO.png)
[](https://i.imgur.com/gbqzSdd.png)
[](https://i.imgur.com/zaw4TIq.png)

**Comment by KINGLIFER** [comment]
Thrir discord is for children. I told them about this a month and a
half ago.  A bit of an immature group. Best to use a different software.

**Comment by illuculent** [comment]
I was able to get past this issue. I renamed the 'venv' folder to '_venv' in the '\stable-diffusion\Packages\Stable Diffusion WebUI\venv'. After the update was successful, I deleted the '_venv' folder. If this is not successful for you, you will be able to delete the newly created 'venv' folder and re-rename the '_venv' back to 'venv' and this should put you back where you were before this attempted fix failed. Good luck!
